src/components/map/map.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Listen,
+  Prop
+} from "@stencil/core";
+import { IComponent } from "../common/interfaces";
+import {
+  Marker,
+  map as LFMap,
+  tileLayer
+  // tslint:disable-next-line:no-submodule-imports
+} from "leaflet/dist/leaflet-src.esm";
+
+@Component({
+  shadow: false,
+  styleUrl: "map.scss",
+  tag: "gx-map"
+})
+export class Map implements IComponent {
+  map: any;
+  @Element() element;
+
+  /**
+   * The coord of initial center of the map.
+   *
+   */
+  @Prop({ mutable: true })
+  center = "0, 0";
+
+  /**
+   * The max zoom level available in the map.
+   *
+   */
+  @Prop() maxZoom = 20;
+
+  /**
+   * The initial zoom level in the map.
+   *
+   */
+  @Prop({ mutable: true })
+  zoom = 1;
+
+  /**
+   * Emmits when the map is loaded.
+   *
+   */
+  @Event() gxMapDidLoad: EventEmitter;
+
+  @Listen("gxMapMarkerDidLoad")
+  onMapMarkerDidLoad(markerInstance) {
+    const markerElement = markerInstance.target;
+    const markerV = markerInstance.detail;
+
+    if (this.map) {
+      markerV.addTo(this.map);
+    } else {
+      this.element.addEventListener("gxMapDidLoad", () => {
+        markerV.addTo(this.map);
+      });
+    }
+    markerElement.addEventListener(
+      "gxMapMarkerDeleted",
+      this.onMapMarkerDeleted.bind(this, markerV)
+    );
+  }
+
+  onMapMarkerDeleted(markerV: Marker) {
+    markerV.remove();
+  }
+
+  componentDidLoad() {
+    const regExp = /^(\-?\d+(\.\d+)?),\s*(\-?\d+(\.\d+)?)$/;
+
+    const elementVar = this.element.querySelector(".gxMap");
+    if (regExp.test(this.center)) {
```

Comment:
What happens if center is changed after the first render? `componentDidLoad` is fired only after the first render of the component. We should also validate coords when the component is updated.
Additionally, we should also validate the coords in the `gx-map-marker` component properties. I think it would be a good idea to refactor this code a little bit so we can have a function to parse and validate coords that we can use in both `gx-map` and `gx-map-marker`.